src/compiler/sourcemap.ts

```diff
@@ -333,6 +408,35 @@ namespace ts {
             }
 
             if (node) {
+                if (isUnparsedSource(node) && node.sourceMapText !== undefined) {
+                    captureSectionalSpanIfNeeded(/*reset*/ true);
+                    const text = node.sourceMapText;
+                    let parsed: {} | undefined;
+                    try {
+                        parsed = JSON.parse(text);
+                    }
+                    catch {
+                        // empty
+                    }
+                    const offset = { line: writer.getLine() - 1, column: writer.getColumn() - 1 };
+                    completedSections.push(parsed
+                        ? {
+                            offset,
+                            map: parsed as SourceMap
+                        }
+                        : {
+                            offset,
+                            // This is just passes the buck on sourcemaps we don't really understand, instead of issuing an error (which would be difficult this late)
+                            url: `data:application/json;charset=utf-8;base64,${base64encode(sys, text)}`

```

Comment:
Would it make sense for us to handle this kind of validation early on when we get the source maps from project references? Instead of storing `sourceMapText` we could have already validated and parsed the source map.